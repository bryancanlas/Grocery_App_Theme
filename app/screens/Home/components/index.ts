export { AddToCartButton } from './AddToCartButton'
export { CategoryCard } from './CategoryCard'
export { CategoryHeader } from './CategoryHeader'
export { Footer } from './Footer'
export { Header } from './Header'
export { HeaderImage } from './HeaderImage'
export { Item } from './Item'
export { Search } from './Search'
export {LogoutModal} from './LogoutModal'