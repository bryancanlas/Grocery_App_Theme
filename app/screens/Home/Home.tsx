import React from 'react';
import { Header, Footer, LogoutModal } from './components';
import { SectionList, View } from '../../components/Themed';
import CartButton from '../../components/CartButton';
import Loading from '../../components/Loading';
import { verticalScale } from '../../constants';
import { useStore } from '../../contexts';
import { LoginModal } from "../../components/LoginModal";


export default function HomeScreen() {
  const { refetchData, loading, networkStatus } = useStore()
  if (loading) return <Loading />

  return (
    <View style={{ flex: 1, justifyContent: 'space-between' }}>
      <SectionList
        contentContainerStyle={{
          paddingBottom: verticalScale(20),
        }}
        bounces={false}
        showsVerticalScrollIndicator={false}
        ListHeaderComponent={<Header />}
        ListFooterComponent={<Footer />}
        sections={[]}
        keyExtractor={(item, index) => item + index}
        renderItem={() => null}
        onRefresh={refetchData}
        refreshing={networkStatus === 7}
      />
      <CartButton />

      {/* Login Modal */}
      
        <LoginModal />
      {/* loginModal end   */}


      {/* Logout Modal */}
           <LogoutModal />
        {/* logout modal end */}
    </View>
  );
}
