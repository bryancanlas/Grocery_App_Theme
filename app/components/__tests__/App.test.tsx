import React from 'react';
import renderer from 'react-test-renderer';

import Home from '../../screens/Home'

test('render correctly', () => {
   const tree = renderer.create(<Home/>).toJSON();
   expect(tree).toMatchSnapshot();
  });