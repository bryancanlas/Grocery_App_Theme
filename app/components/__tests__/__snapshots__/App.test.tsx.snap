// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render correctly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#fff",
        "borderColor": "#e0e0e0",
        "shadowColor": "#000",
      },
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#fff",
          "borderColor": "#e0e0e0",
          "shadowColor": "#000",
        },
        Object {
          "backgroundColor": "#9291a1",
          "height": "100%",
          "opacity": 0.5,
          "position": "absolute",
          "width": "100%",
        },
      ]
    }
  />
  <Image
    source={1}
    style={
      Object {
        "borderRadius": 78.57142857142857,
        "height": 157.14285714285714,
        "position": "absolute",
        "transform": Array [
          Object {
            "rotate": "0deg",
          },
        ],
        "width": 157.14285714285714,
      }
    }
  />
</View>
`;
